### ISSUE TITLE
Super structured output via MLX

### ISSUE THREAD

**Issue by ashtonmeuser** [mi12labs/SwiftAI#16]
Tracking issue for structured output when using MLX model.

**Comment by mi12-root** [comment]
In theory we have access to logits during inference, so it's possible to implement constrained decoding in SwiftAI directly. If there are a production use cases for it, I can consider prioritizing it in favor of other feature requests.
